src/context/AppContextProvider.tsx

```diff
@@ -76,6 +77,7 @@ export const AppContextProvider = ({
       if (!operation) {
         throw new Error(`Unknown operation type: ${operationName}`);
       }
+      setIsOperationValidated(false);
```

Comment:
might be safer to move this up to be the first thing this function does, before an error is potentially thrown. because if an error is thrown before we set the operation to false, it would still show true